src/Foundatio.Repositories.Elasticsearch/Repositories/ElasticReadOnlyRepositoryBase.cs

```diff
@@ -426,7 +426,7 @@ public abstract class ElasticReadOnlyRepositoryBase<T> : ISearchableReadOnlyRepo
             result = response.Hits.Select(h => h.ToFindHit()).ToList();
 
             if (IsCacheEnabled && options.ShouldUseCache())
-                await AddDocumentsToCacheAsync(result, options).AnyContext();
+                await AddDocumentsToCacheAsync(result, options, options.GetConsistency(DefaultConsistency) == Consistency.Immediate ? false : true).AnyContext();
```

Comment:
Seems kinda odd to me that I'd use a consistency mode when doing a search. Seems like the common behavior would be to cache the results on add/save or pass a consistency mode when calling add/save.

Reply:
It's not odd. It's pretty common to say that I want to do something like a FindOne using a query and that I want to use immediate consistency meaning that I will wait for any pending writes.

Reply:
We've never set `Consistency.Immediate` in any of our apps on a FindOne call to my knowledge.

Reply:
During auth we often lookup by email address using a FindOne with immediate consistency.